tests/FSharp.Test.Utilities/CompilerAssert.fs

```diff
@@ -556,6 +556,27 @@ let main argv = 0"""
 
             Assert.AreEqual(errorsExpectedBaseLine.Replace("\r\n","\n"), errorsActual.Replace("\r\n","\n"))
 
+    static member TypeCheckWithOptionsAndName options name (source: string) =
```

Comment:
Can we not do this as a compiler type with the fluent Apis?

Reply:
It is used by the IL verification only. I can have it as separate function for new api as well. 
So far "typecheck" function was sufficient enough for our needs.

Reply:
Yeah, but I dislike having the TypeCheckWithOptionsAndName and extending it whenever we think of adding some new thing to it.  The fluent thing with all the data is kinda much cleaner, and the nicest thing about your framework.